addons/block_code/types/types.gd

```diff
@@ -3,7 +3,7 @@ extends Node
 
 enum BlockType {
 	NONE,
-	EXECUTE,
```

Comment:
Ah I don't know Unreal "execute". Can you explain what does the execute type mean? There is nothing in GDScript that maps to it. I was even considering renaming "entry" to "(virtual) method".

The difference between block types and block classes is a bit confusing to me. But maybe you can provide examples of:
- a potential basic block that's not an entry block
- a potential statement block that's not a statement block

I would like to hear input from @dylanmccall as well.

Reply:
The only thing that `block_type` does is dictate which blocks can be snapped to a certain `SnapPoint`. The `SnapPoint` also has a `block_type`, and only blocks with matching `block_type` can be snapped. So, an `EXECUTE` block will only snap to the inside of control blocks, or the end of other statement blocks. A `STRING` block will only snap to parameter inputs of `STRING` type.

The block class is associated with the scene (see `get_scene_path()`, and is a template for building whatever block you may want. Say you need a block that has a specific name and label, but doesn't generate code, then you would instantiate and customize a `BasicBlock`. But if you needed a block that generates code from parameters then you could use a `StatementBlock`. What if you wanted to have an entry point with parameters (like the `on signal` block). Then you could instantiate the `StatementBlock` and give it the `ENTRY` type so it will not snap to `EXECUTE` snap points.

Execution is just a way of describing the flow of instructions in the program. The program starts executing at the entry block, then moves on to the next instruction/statement (block) it needs to execute. That's why I named these blocks that are supposed to be executed in this order `EXECUTE`. Snapping a block to the bottom of another is essentially saying to execute it next.

The reason to not rename `BasicBlock` is because there is no reason to specialize it as an entry block, when it could have (albeit very limited) other purposes. You can specialize it in `CategoryFactory`.

Reply:
There is another exception! For the body entered node (or even the process nodes if we want to include `delta`), we want to have a parameter block that is copyable in the scope of the following script. The code for the body entered node looks like this:
```
b = BLOCKS["statement_block"].instantiate()
b.block_type = Types.BlockType.ENTRY
b.block_format = "On body [body: NODE] entered"
test_list.append(b)
```
With the same logic that creates parameter inputs, you can create these copyable function parameters, where you use square brackets instead of curlies. There are definitely other ways of splitting this functionality up away from `StatementBlock`, but for the MVP I think this is acceptable.